examples/albef/data/retrieval_datamodule.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Tuple
+
+import torch
+from examples.albef.data.retrieval_dataset import (
+    ImageToTextRetrievalDataset,
+    RetrievalTrainingDataset,
+    TextToImageRetrievalDataset,
+)
+from examples.albef.data.transforms import (
+    ALBEFTextTransform,
+    testing_image_transform,
+    training_image_transform,
+)
+from pytorch_lightning import LightningDataModule
+from torch import Tensor
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import DataLoader, Dataset, DistributedSampler
+
+
+class RetrievalDataModule(LightningDataModule):
```

Comment:
One high-level comment here: imo the advantages of datamodule are less obvious when we have non-standard splits (i.e. not train/val/split). In that case it may be better to just construct the dataloaders directly without the datamodule wrapper